substratest/assets.py

```diff
@@ -50,17 +60,60 @@ def get(self):
         return self._asset
 
 
-class _FutureMixin(abc.ABC):
+class ComputePlanFuture(BaseFuture):
+    def __init__(self, asset, session):
+        self._asset = asset
+        self._getter = session.get_compute_plan
+        self._get_traintuple = session.get_traintuple
+        self._get_composite_traintuple = session.get_composite_traintuple
+        self._get_aggregatetuple = session.get_aggregatetuple
+        self._get_testtuple = session.get_testtuple
+
+    def wait(self, timeout=FUTURE_TIMEOUT):
+        """wait until all tuples are completed (done or failed)."""
+        for key in self._asset.traintuple_keys:
+            self._get_traintuple(key).future().wait(timeout, raises=False)
+        for key in self._asset.composite_traintuple_keys:
+            self._get_composite_traintuple(key).future().wait(timeout, raises=False)
+        for key in self._asset.aggregatetuple_keys:
+            self._get_aggregatetuple(key).future().wait(timeout, raises=False)
+        for key in self._asset.testtuple_keys:
+            self._get_testtuple(key).future().wait(timeout, raises=False)
```

Comment:
The tuples should be ordered by execution order: it will avoid a timeout on the first one. To do that we could use the rank.